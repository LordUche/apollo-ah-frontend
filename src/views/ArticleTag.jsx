import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
/**
 * @returns {JSX} - The JSX representation of the article's tags
 *
 */
function Tag({ tag }) {
  return (
    <Link to={`/search?tag=${tag.tagName}`} className="badge">
      {tag.tagName}
    </Link>
  );
}
Tag.propTypes = {
  tag: PropTypes.object.isRequired,
};

export default Tag;
