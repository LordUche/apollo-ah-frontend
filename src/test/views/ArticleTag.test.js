import React from 'react';
import { shallow } from 'enzyme';
import ArticleTag from '../../views/ArticleTag';

const tag = {
  tagName: 'Technology',
};

describe('<ArticleTag>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<ArticleTag tag={tag} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('badge')).toBe(true);
  });
});
