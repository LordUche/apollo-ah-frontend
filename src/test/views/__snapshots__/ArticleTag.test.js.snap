// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleTag> should render without crashing 1`] = `
<Link
  className="badge"
  replace={false}
  to="/search?tag=Technology"
>
  Technology
</Link>
`;
